transcript bactopia (#1402)
nf-core/nf-co.re: markdown/events/2022/bytesize-bactopia.md

This week, Robert A. Petit III ([@rpetit3](https://github.com/rpetit3)) will introduce us to his pipeline [Bactopia](https://github.com/bactopia/bactopia). A first for bytesize, bactopia is not an nf-core pipeline but is rather [influenced by nf-core](https://bactopia.github.io/acknowledgements/#nf-core).

Robert will show us the pipeline and explain how he is using nf-core components (such as [modules](https://nf-co.re/modules)) in this pipeline, despite it being outside the main nf-core organisation.

<details markdown="1"><summary>Video transcription</summary>
**Note: The content has been edited for reader-friendliness**

[0:01](https://www.youtube.com/watch?v=egjgcmeJ0wQ&t=1)

> Hello everyone my name is Franziska Bonath, I'm the today's host for the bite size talk and with me is Robert Petit. He is from the Wyoming public health laboratory and is going to talk about bactopia which is not part of nf-core but it's using nf-core components and nf-core pipelines, so off to you!

Reviewer: ```suggestion
(host) Hello everyone my name is Franziska Bonath, I'm the today's host for the bytesize talk and with me is Robert Petit. He is from the Wyoming public health laboratory and is going to talk about bactopia which is not part of nf-core but it's using nf-core components and nf-core pipelines, so off to you!
```

[0:28](https://www.youtube.com/watch?v=egjgcmeJ0wQ&t=28)

> I'm so sorry I did mute you by mistake. You can unmute yourself. I'm sure...

Reviewer: ```suggestion
(host) I'm so sorry I did mute you by mistake. You can unmute yourself. I'm sure...
```

There we go!

> I'm so sorry!

Reviewer: ```suggestion
(host) I'm so sorry!
```

No don't worry about it!

[0:50](https://www.youtube.com/watch?v=egjgcmeJ0wQ&t=50)
Thank you for having me. I'm super excited to be here today. I'll just be introducing you to bactopia. I'll give you a few use cases of how I'm using

> nf-core components in bactopia. But first a little motivation on how bactopia came up. Over the last 10 years we saw a a quite nice growth in the publicly available bacterial genomes. From ena sra ddbj we went from about 7500 in 2010 to about 1.5 million. While that pales in comparison to COVID, which I think is 12 million plus now, for bacteria that is quite a lot. Over these last 10 years we've also seen the rise of containers and package managers, such as docker, singularity and bioconda. And then workflow managers and the curators behind those, such as nextflow and nf-core. At 2010 maybe we couldn't use all the data but it really starts in 2022, it really makes you wonder: can we make use of all these publicly developed genomes? In 2010 I remember, passing tar balls with binaries and then email the sequencing instrument groups say "can I get a binary to your assembler?" and stuff like that. There was no real... It was just "make install" and hope it installs.

Author: Yes, I wasn't sure what he meant, so I didn't change too much...

[5:09](https://www.youtube.com/watch?v=egjgcmeJ0wQ&t=309)
Some design principles behind bactopia. One bactopia requires all tools that are included in it to be available from bioconda. The main reasons is because it's 2022. People shouldn't have to figure out how to install a tool now. It should be able to either use a container or some sort of conda. It should be an easy, simple process and because bioconda has the downstream containerization, so every recipe gets a docker container through bio-containers and a singularity image through galaxy project, we have all those tools necessary to start using these immediately. I also require all modules and bactopia tools to also be available from nf-core modules. If it's not there we add it. Bactopia should be easy to install and adaptable to the user's needs. Converting to DSL2 has made this much, much easier.

[6:14](https://www.youtube.com/watch?v=egjgcmeJ0wQ&t=374)

> There are three sides to bactopia. You can think of these as checkpoints between the three. There's bactopia helpers, bactopia and bactopia tools. The bactopia helpers help you get started using bactopia. These are your pre-analysis steps or some commands to post analysis get information. One is the bactopia citations, which will print out citations for all the tools used by bactopia. The bactopia datasets command allows you to go and download publicly available data sets, that can supplement your analysis. These include things like refseq and genbank sketches, as well as pub msc schemas and many more. The bactopia download command will pre-build conda environments for you, pool docker containers or download singularity images as a pre-step. That way you're not doing that while you're starting to process the nextflow. The bactopia preparer will create a file of file name similar to the sample sheet that you see in many of the nf-core pipelines. This allows you to really process as many genomes as you want. The bactopia search, one of my favorite, it takes a query, queries ena's api, then returns a list of experiment accessions that you can then feed to bactopia to download and start processing.

Author: nice to see all the stuff the I didn't really know what they are to be corrected :) 👍

[16:37](https://www.youtube.com/watch?v=egjgcmeJ0wQ&t=997)
Thank you and I will take any questions folks have.
(host) Thank you very much. I have enable now for everyone to unmute themselves, if they have questions. Otherwise we can start with the one question that is already in the chat, which is from Olaitan. He is asking: "it seems like bactopia is not in the APT repository, could you work on including it?"
(answer) I don't know much about including tools in the apt repository. It is available from bioconda, so you can "conda install" bactopia. I think there would be many components of bactopia, that aren't in the APT repository, so I don't know how that would work. It would be something you would have to add all the dependencies to the APT repository, and I think the time required for that and the learning, I don't have the bandwidth for at the moment. Definitely consider using the bioconda install and then from there you can use conda docker singularity.

[17:55](https://www.youtube.com/watch?v=egjgcmeJ0wQ&t=1075)
(host) okay thank you. Are there any more questions from the audience?

> (question) I do have a follow-up question. It's not really based on the APT repository thing, it's the the workflow. In terms of the different steps, that bactopia does. I didn't get what Robert meant by the final step that he talked about. Something about the analysis. I was wondering what are the specific things, like what are you measuring at the end of the day with bactopia? Specifically in terms of the omics analysis.

Reviewer: ```suggestion
(question) I do have a follow-up question. It's not really based on the APT repository thing, it's the workflow. In terms of the different steps, that bactopia does. I didn't get what Robert meant by the final step that he talked about. Something about the analysis. I was wondering what are the specific things, like what are you measuring at the end of the day with bactopia? Specifically in terms of the omics analysis.
```

> (answer) It's going to include pretty much all your standard bacterial genomics. You're going to qc the reads, how well did you sequence your sample, what's the average read length, all that fun stuff. Then it's going to characterize your sample. what nst schema? Does it have certain antibiotic resistance genes that you may be interested in? Does it have snips and endels against a reference genome that you selected? How does it compare to public available genomes, does it look like what you expected? If you thought you sequenced the staph aureus, and you know it came up as looking more like enterococcus, that's something. Those are the type of analysis results. On the bactopia documentation there's an overview of the workflow at each step and then output overviews on all the output files that you get for both bactopia and all the bactopia tools. Those output give you a description of what's in the each of the files.

Reviewer: ```suggestion
(answer) It's going to include pretty much all your standard bacterial genomics. You're going to QC the reads, how well did you sequence your sample, what's the average read length, all that fun stuff. Then it's going to characterize your sample. what MLST schema? Does it have certain antibiotic resistance genes that you may be interested in? Does it have SNPs and INDELs against a reference genome that you selected? How does it compare to publicly available genomes, does it look like what you expected? If you thought you sequenced the Staphylococcus aureus, and you know it came up as looking more like Enterococcus, that's something. Those are the type of analysis results. On the bactopia documentation there's an overview of the workflow at each step and then output overviews on all the output files that you get for both bactopia and all the bactopia tools. Those output give you a description of what's in the each of the files.
```

[19:52](https://www.youtube.com/watch?v=egjgcmeJ0wQ&t=1192)

> (question) Okay thanks! My final question is, I saw you integrated this with illumina reads and I believe nanopore reads. What happened to pacbio?

Reviewer: ```suggestion
(question) Okay thanks! My final question is, I saw you integrated this with Illumina reads and I believe Nanopore reads. What happened to PacBio?
```

> (answer) Honestly I just haven't been exposed to pacbio data much so far in my analyses and my studies. I think if I start using pacbio, then pacbio data will come in, otherwise I think I would need support from the community to add that type. Just because I don't have the opportunity to use it on a daily basis like I do illumina and nanopore. We need someone else to help out there.

Reviewer: ```suggestion
(answer) Honestly I just haven't been exposed to PacBio data much so far in my analyses and my studies. I think if I start using PacBio, then PacBio data will come in, otherwise I think I would need support from the community to add that type. Just because I don't have the opportunity to use it on a daily basis like I do Illumina and Nanopore. We need someone else to help out there.
```

(from audience) cool, thanks! Great job! thank you!
(host) thank you very much, are there any other questions?

[20:55](https://www.youtube.com/watch?v=egjgcmeJ0wQ&t=1255)

> (host) I don't see anything pop up so, I would like to thank you again, Robert. Everyone else, there is always the chance to ask more questions, if they come later, at the bite size channel on slack. I guess you can also contact Robert directly and this video will also be uploaded to youtube. I would like to thank, apart from Robert of course, the Chan Zuckerberg Initiative, who is funding these talks, and thank you everyone for joining in. Thank you.

Reviewer: ```suggestion
(host) I don't see anything pop up so, I would like to thank you again, Robert. Everyone else, there is always the chance to ask more questions, if they come later, at the bytesize channel on Slack. I guess you can also contact Robert directly and this video will also be uploaded to Youtube. I would like to thank, apart from Robert of course, the Chan Zuckerberg Initiative, who is funding these talks, and thank you everyone for joining in. Thank you.
```
